test/network_analysis/conservation_laws.jl

```diff
@@ -127,20 +127,22 @@ let
     @test osol1[sps] ≈ osol2[sps] ≈ osol3[sps]
 
     # Checks that steady states found using nonlinear solving and steady state simulations are identical.
-    nsys = complete(convert(NonlinearSystem, rn; remove_conserved = true, remove_conserved_warn))
-    nprob1 = NonlinearProblem{true}(nsys, u0, p)
-    nprob2 = NonlinearProblem(rn, u0, p)
-    nprob3 = NonlinearProblem(rn, u0, p; remove_conserved = true, remove_conserved_warn)
-    ssprob1 = SteadyStateProblem{true}(osys, u0, p)
-    ssprob2 = SteadyStateProblem(rn, u0, p)
-    ssprob3 = SteadyStateProblem(rn, u0, p; remove_conserved = true, remove_conserved_warn)
-    nsol1 = solve(nprob1, NewtonRaphson(); abstol = 1e-8)
-    # Nonlinear problems cannot find steady states properly without removing conserved species.
-    nsol3 = solve(nprob3, NewtonRaphson(); abstol = 1e-8)
-    sssol1 = solve(ssprob1, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
-    sssol2 = solve(ssprob2, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
-    sssol3 = solve(ssprob3, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
-    @test nsol1[sps] ≈ nsol3[sps] ≈ sssol1[sps] ≈ sssol2[sps] ≈ sssol3[sps]
+    @test_broken begin # Conservation law removal currently not working for NonlinearSystems due to MTK depricating something. https://github.com/SciML/ModelingToolkit.jl/issues/3458, https://github.com/SciML/ModelingToolkit.jl/issues/3411
+        nsys = complete(convert(NonlinearSystem, rn; remove_conserved = true, remove_conserved_warn))
+        nprob1 = NonlinearProblem{true}(nsys, u0, p; guesses = [nsys.Γ => [0.0]])
+        nprob2 = NonlinearProblem(rn, u0, p)
+        nprob3 = NonlinearProblem(rn, u0, p; remove_conserved = true, remove_conserved_warn)
+        ssprob1 = SteadyStateProblem{true}(osys, u0, p)
+        ssprob2 = SteadyStateProblem(rn, u0, p)
+        ssprob3 = SteadyStateProblem(rn, u0, p; remove_conserved = true, remove_conserved_warn)
+        nsol1 = solve(nprob1, NewtonRaphson(); abstol = 1e-8)
+        # Nonlinear problems cannot find steady states properly without removing conserved species.
+        nsol3 = solve(nprob3, NewtonRaphson(); abstol = 1e-8)
+        sssol1 = solve(ssprob1, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
+        sssol2 = solve(ssprob2, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
+        sssol3 = solve(ssprob3, DynamicSS(Tsit5()); abstol = 1e-8, reltol = 1e-8)
+        @test nsol1[sps] ≈ nsol3[sps] ≈ sssol1[sps] ≈ sssol2[sps] ≈ sssol3[sps]
+    end
```

Comment:
`NonlinearSystem` + conservation laws is broken, and requires MTK fixes before we can make it work.

Reply:
yes, although there is hope I might be able to fix it now when a bug got fixed.